### ISSUE TITLE
Memory leak in BCObservable

### ISSUE THREAD

**Issue by fredRos** [bat/bat#250]
Why is `fObservableValue` a pointer? Why is there an option in the ctor to share it? It is not deleted, presumably because it could be shared and that would lead to a double delete.

I don't understand why sharing is needed. If it really is, it should use a `std::shared_ptr`. I don't want to have a leak here

```c++
BCObservable::BCObservable(const BCObservable& other, bool share_pointer)
    : BCVariable(other)
    , fObservableValue(share_pointer ? other.fObservableValue : new double)
{
}

BCObservable::~BCObservable()
{
}
```

**Comment by fredRos** [comment]
For reference: I found this with valgrind but it is actually easier to spot with gcc's sanitizer as it emits less noise about ROOT leaks. I changed `example/Makefile-common.am`:

```make
AM_CXXFLAGS = \
	-g -O0 -fsanitize=address \
	-I$(top_builddir)/src/.includes \
	-I$(top_builddir)/models/base/.includes \
	-I$(top_builddir)/models/mtf/.includes \
	$(ROOTAUXCFLAGS) \
	$(ROOTCFLAGS)
```

**Comment by greenwald** [comment]
It seems to be used by the `BCPriorModel`. But I think I can avoid this by having `BCPriorModel` simply copy the values. I'll investigate.

**Comment by greenwald** [comment]
OK. In fact `BCPriorModel` is redundantly already copying the values in `BCPriorModel::CalculateObservables`:

```
fModel.CalculateObservables(parameters);
for (unsigned i = 0; i < GetNObservables(); ++i)
    GetObservable(i).Value(fModel.GetObservable(i).Value());
```
With the sharing of pointers, only the first line is necessary. I will remove the sharing of pointers and keep the rest of the code as is.

**Comment by fredRos** [comment]
fixed by #251
